prime-dotnet-webapi/Services/CommunitySiteService.cs

```diff
@@ -196,40 +196,41 @@ private void UpdateBusinessHours(Site current, CommunitySiteUpdateModel updated)
 
         private void UpdateRemoteUsers(Site current, CommunitySiteUpdateModel updated)
         {
-            // Wholesale replace the remote users
-            foreach (var remoteUser in current.RemoteUsers)
+            if (updated.RemoteUsers == null)
             {
-                foreach (var certification in remoteUser.RemoteUserCertifications)
-                {
-                    _context.Remove(certification);
-                }
-                _context.RemoteUsers.Remove(remoteUser);
+                return;
             }
 
-            if (updated?.RemoteUsers != null && updated?.RemoteUsers.Count() != 0)
+            foreach (var updateRemoteUser in updated.RemoteUsers)
             {
-                foreach (var remoteUser in updated.RemoteUsers)
-                {
-                    remoteUser.SiteId = current.Id;
-                    var remoteUserCertifications = new List<RemoteUserCertification>();
+                var existingRemoteUser = _context.RemoteUsers
+                    .FirstOrDefault(u => u.Id == updateRemoteUser.Id);
 
-                    foreach (var certification in remoteUser.RemoteUserCertifications)
+                if (existingRemoteUser == null)
+                {
+                    current.RemoteUsers.Add(updateRemoteUser);
+                }
+                else
+                {
+                    _context.Entry(existingRemoteUser).CurrentValues.SetValues(updateRemoteUser);
```

Comment:
This line will overwrite the SiteId on the existing `RemoteUser` with whatever SiteId comes from the UpdateModel, which is bad since it could have been set by the FE to any value. We need to set the SiteId on the UpdateModel to the site's Id to prevent it from being manipulated.